### PR TITLE
[Refactor] Unify GetTable/DB into simplified function MetadataMgr.getXXX

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by Seaven — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Review by satanson — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java [hunk 1]
```diff
@@ -178,18 +178,20 @@ public Optional<ConnectorMetadata> getOptionalMetadata(String catalogName) {
      * @param catalogName catalog's name
      * @return ConnectorMetadata
      */
-    public Optional<ConnectorMetadata> getOptionalMetadata(Optional<String> queryId, String catalogName) {
+    public static Optional<ConnectorMetadata> getOptionalMetadata(Optional<String> queryId, String catalogName) {
         if (Strings.isNullOrEmpty(catalogName) || CatalogMgr.isInternalCatalog(catalogName)) {
-            return Optional.of(localMetastore);
+            return Optional.of(GlobalStateMgr.getCurrentState().getLocalMetastore());
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java:183]: There is no need to replace this class

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java [hunk 2]
```diff
@@ -178,18 +178,20 @@ public Optional<ConnectorMetadata> getOptionalMetadata(String catalogName) {
...
 
-        CatalogConnector connector = connectorMgr.getConnector(catalogName);
+        CatalogConnector connector = GlobalStateMgr.getCurrentState().getConnectorMgr().getConnector(catalogName);
         if (connector == null) {
             LOG.error("Failed to get {} catalog", catalogName);
             return Optional.empty();
         }
 
         if (queryId.isPresent()) { // use query-level cache if from query
+            LoadingCache<String, QueryMetadatas> metadataCacheByQueryId =
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java:193]: ditto

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java [hunk 3]
```diff
@@ -178,18 +178,20 @@ public Optional<ConnectorMetadata> getOptionalMetadata(String catalogName) {
      * @param catalogName catalog's name
      * @return ConnectorMetadata
      */
-    public Optional<ConnectorMetadata> getOptionalMetadata(Optional<String> queryId, String catalogName) {
+    public static Optional<ConnectorMetadata> getOptionalMetadata(Optional<String> queryId, String catalogName) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java:181]: why modify it to static?

**Review by stephen-shelby — COMMENTED** [review]
We should not destroy MetadataMgr encapsulation

**Pull request closed without merging**